conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -162,3 +186,38 @@ def get_find_mode(self, dep):
         if tmp is None:
             return "config"
         return tmp.lower()
+
+    def generate_aggregator(self):
+        host = self._conanfile.dependencies.host
+        build_req = self._conanfile.dependencies.direct_build
+        test_req = self._conanfile.dependencies.test
+
+        configs = []
+        for require, dep in list(host.items()) + list(build_req.items()) + list(test_req.items()):
+            if not require.direct:
+                continue
+            if require.build and dep.ref.name not in self.build_context_activated:
+                continue
+            cmake_find_mode = self.get_property("cmake_find_mode", dep)
+            cmake_find_mode = cmake_find_mode or FIND_MODE_CONFIG
+            cmake_find_mode = cmake_find_mode.lower()
+            find_module_mode = True if cmake_find_mode == FIND_MODE_MODULE else False
+            config = ConfigTemplate(self, require, dep, find_module_mode)
+            configs.append(config)
+
+        template = textwrap.dedent("""\
+            message(STATUS "Conan: Using CMakeDeps conandeps.cmake aggregator via include()")
+            message(STATUS "Conan: It is recommended to use explicit find_package() per dependency instead")
+
+            {% for config in configs %}
+            find_package({{config.file_name}})
+            {% endfor %}
+
+            add_library(conandeps INTERFACE)
```

Comment:
Consider using a variable like the original generator, where the value of the variable is a list of targets
https://docs.conan.io/1/integrations/build_system/cmake/cmake_generator.html#global-variables-approach

(I'd also consider using a name like `CONAN_DEPS_LEGACY` or something like that)